cloud/pod_util.go

```diff
@@ -335,3 +356,102 @@ func exportPodEnvVars(smConf evergreen.SecretsManagerConfig, opts pod.TaskContai
 
 	return allEnvVars
 }
+
+// GetFilteredResourceIDs gets at most n resources that match the given resource
+// and tag filters. If n is not a positive integer, the results are not limited.
+func GetFilteredResourceIDs(ctx context.Context, c cocoa.TagClient, resources []string, tags map[string][]string, limit int) ([]string, error) {
+	if limit == 0 {
+		return []string{}, nil
+	}
+	if limit < 0 {
+		limit = math.MaxInt64
+	}
+
+	var tagFilters []*resourcegroupstaggingapi.TagFilter
+	for key, vals := range tags {
+		tagFilters = append(tagFilters, &resourcegroupstaggingapi.TagFilter{
+			Key:    aws.String(key),
+			Values: utility.ToStringPtrSlice(vals),
+		})
+	}
+	resourceFilters := utility.ToStringPtrSlice(resources)
+
+	var allIDs []string
+	remaining := limit
+	for ids, nextToken, err := getResourcesPage(ctx, c, resourceFilters, tagFilters, nil, limit); ; ids, nextToken, err = getResourcesPage(ctx, c, resourceFilters, tagFilters, nextToken, remaining) {
```

Comment:
Can we break this up into two lines for readability?

Reply:
Hmm, it's hard to break up into two separate lines because the initialization condition is different from the repeated iteration condition. I tried to modify it to be more readable.

Reply:
Thanks!